hive/src/main/java/org/elasticsearch/hadoop/hive/pushdown/TreeParser.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.hadoop.hive.pushdown;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.elasticsearch.hadoop.hive.pushdown.node.Node;
+import org.elasticsearch.hadoop.hive.pushdown.node.OpNode;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Parse The structure of the tree parse ,and get a new type about T.
+ *
+ * @param <T>
+ */
+public abstract class TreeParser<T> implements Serializable {
```

Comment:
The logic for this is split across two classes, but I don't see any situations where we would need to use a different subclass of TreeParser. Can we simplify here?

Reply:
done, same reason as above